### ISSUE TITLE
Rename blitable

### ISSUE THREAD

**Issue by jacereda** [carp-lang/Carp#1114]
My Amiga past makes me expect more than 'copy' when I read 'blit':

https://en.wikipedia.org/wiki/Blitter

**Comment by jacereda** [comment]
I mean, you can't blit floating point values (or pointers, like I added to the `blit` interface in #1111 )

**Comment by scolsen** [comment]
I would also like to rename it too, `blit` feels a bit too obscure to me. Could we perhaps use `flat` or `unboxed`? Possibly also too obscure, but the notion would be that it's the opposite of a "boxed" value, which typically denotes something on the heap, and denotes something that's stack allocated, which is in keeping w/ Erik's original thoughts on what `blit` indicates:

> With "freely copyable" I mean anything that only lives on the stack and where copying or deleting is taken care of by the C value type semantics.

There's already precedence for such terminology in the ATS programming language which differentiates between allocation of "flat" values and allocation of "boxed" values.

```ats
(* Flat Tuples *)

val xyx = ('A', 1, 2.0)

(* Boxed tuples, pointer to heap allocated values *)
(* spaces are required to disambiguate *)
val xyx-box = '( 'A', 1, 2.0 ) 
```

http://ats-lang.sourceforge.net/DOCUMENT/INT2PROGINATS/HTML/x302.html

**Comment by eriksvedang** [comment]
`Flat` is nice, I like it!

**Comment by TimDeve** [comment]
Not that I have any better name to propose but isn't it a bit weird to call something Flat if we don't have Box?

**Comment by scolsen** [comment]
I think `mcpy` would work as well, I agree that `flat` would be nicer if `box` was a more prevalent concept in our code and docs.

**Comment by jacereda** [comment]
What about `trivially-copyable`?

**Comment by eriksvedang** [comment]
> Not that I have any better name to propose but isn't it a bit weird to call something `Flat` if we don't have `Box`?
> 
> 
> 
> Is `mcpy` too confusing?

We might add Box later though! I think the most important thing is that the name is good on its own.

`mcpy` is a little too hard to read/say for my liking – in that case I think it makes more sense to just call it `memcpy` to make the correlation with the C function super clear.

@jacereda That name is pretty interesting too, it's very clear. 

Maybe `shallow` could work?

**Comment by TimDeve** [comment]
`shallow-copy`?

**Comment by eriksvedang** [comment]
Yes, that too. If seen as a "tag", the interface should probably just be called `shallow` but if we're going to actually call it, `shallow-copy` is better (and this argument applies to all other name suggestions too).

**Comment by jacereda** [comment]
> @jacereda That name is pretty interesting too, it's very clear.

It's what C++ uses: https://en.cppreference.com/w/cpp/types/is_trivially_copyable

**Comment by eriksvedang** [comment]
I'll switch to `trivially-copyable` with the single function `trivial-copy`.
